spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -173,28 +173,85 @@ An example of an active channel on the controller chain can look like this:
-
-#### Version negotiation summary
+#### **Metadata negotiation summary**
 
 `interchain-account-address` is the address of the interchain account registered on the host chain by the controller chain.
 
-| Initiator | Datagram         | Chain acted upon |  Version (Controller, Host) |
-| --------- | ---------------- | ---------------- |  ---------------------- | 
-| Controller| ChanOpenInit     | Controller       |  (ics27-1, none)        | 
-| Relayer   | ChanOpenTry      | Host             |  (ics27-1, ics27-1.{interchain-account-address})        | 
-| Relayer   | ChanOpenAck      | Controller       |  (ics27-1, ics27-1.{interchain-account-address})        | 
-| Relayer   | ChanOpenConfirm  | Host             |  (ics27-1, ics27-1.{interchain-account-address})        | 
+**INIT**
+
+Initiator: Controller
+
+Datagram: ChanOpenInit
+
+Chain Acted Upon: Controller
+
+Version: 
+```json
+{
+  "Version": "ics27-1",
+  "Encoding": "requested_encoding_type",
+  "TxType": "requested_transaction_type",
+  "ControllerConnectionId": "self_connection_id",
+  "HostConnectionId": "counterparty_connection_id",
+  "Address": ""
+}
+```
+
+Comments: The address is left empty since this will be generated and relayed back by the host chain.
+
+**TRY**
+
+Initiator: Relayer
+
+Datagram: ChanOpenTry
+
+Chain Acted Upon: Host
+
+Version: 
+```json
+{
+  "Version": "ics27-1",
+  "Encoding": "negotiated_encoding_type",
+  "TxType": "negotiated_TxType",
+  "ControllerConnectionId": "counterparty_connection_id",
+  "HostConnectionId": "self_connection_id",
+  "Address": "interchain_account_address"
+}
+```
+
+Comments: The ICS-27 application on the host chain is responsible for returning this version given the counterparty version set by the controller chain in INIT. The host chain must agree with the single encoding type and a single tx type that is requested by the controller chain (ie. included in counterparty version). If the requested encoding or tx type is not supported, then the host chain must return an error and abort the handshake.
+The host chain must also generate the interchain account address and populate the address field in the version with the interchain account address string.
+
+**ACK**
 
-The channel `version` string passed into the `OnChanOpenTry` callback by a relayer will contain the account address of the interchain account that will be registered on the host chain. A relayer will then pass this version into the `OnChanOpenAck` step on the controller chain as the `counterpartyVersion`. The controller chain will then proceed to parse the interchain account address from the `counterpartyVersion` string and store this address in state with a map to the associated portID.  
+Initiator: Relayer
+
+Datagram: ChanOpenAck
+
+Chain Acted Upon: Controller
+
+CounterpartyVersion: 
+```json
+{
+  "Version": "ics27-1",
+  "Encoding": "negotiated_encoding_type",
+  "TxType": "negotiated_TxType",
```

Comment:
nit: there's an inconsistency between:
`negotiated_encoding_type`
`negotiated_TxType`

maybe use `negotiated_tx_type` since the ConnectionId's use `connection_id`